packages/ui/Scroller/Scroller.js

```diff
@@ -27,128 +27,95 @@ import css from './Scroller.module.less';
  * @ui
  * @public
  */
-class ScrollerBase extends Component {
-	static displayName = 'ui:ScrollerBase'
-
-	static propTypes = /** @lends ui/Scroller.ScrollerBase.prototype */ {
-		children: PropTypes.node.isRequired,
-
-		/**
-		 * Callback method of scrollTo.
-		 * Normally, `Scrollable` should set this value.
-		 *
-		 * @type {Function}
-		 * @private
-		 */
-		cbScrollTo: PropTypes.func,
-
-		/**
-		 * Direction of the scroller.
-		 *
-		 * Valid values are:
-		 * * `'both'`,
-		 * * `'horizontal'`, and
-		 * * `'vertical'`.
-		 *
-		 * @type {String}
-		 * @default 'both'
-		 * @public
-		 */
-		direction: PropTypes.oneOf(['both', 'horizontal', 'vertical']),
-
-		/**
-		 * Prop to check context value if Scrollbar exists or not.
-		 *
-		 * @type {Boolean}
-		 * @private
-		 */
-		isVerticalScrollbarVisible: PropTypes.bool,
-
-		/**
-		 * `true` if RTL, `false` if LTR.
-		 *
-		 * @type {Boolean}
-		 * @private
-		 */
-		rtl: PropTypes.bool
-	}
-
-	static defaultProps = {
-		direction: 'both'
-	}
-
-	constructor (props) {
-		super(props);
-
-		this.containerRef = React.createRef();
-	}
-
-	componentDidMount () {
-		this.calculateMetrics();
-	}
-
-	componentDidUpdate (prevProps) {
-		this.calculateMetrics();
-		if (this.props.isVerticalScrollbarVisible && !prevProps.isVerticalScrollbarVisible) {
-			this.forceUpdate();
+let ScrollerBase = (props, reference) => {
```

Comment:
What about applying `forwardRef` here rather than re-allocate `ScrollerBase`?

Reply:
I fixed `scroller`, `scrollable`, `scrollableNative`.
But, in case of `VirtuaListBase`, `VirtualListCore function` is located in `VirtualListBaseFactory function`.
So I cannot fix the `VirtuaListBase` due to lint error  "React Hook \"useRef\" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.", 

I think we need to use curry instead of  `VirtualListBaseFactory function` .
I hope we could do that on PLAT-98957